src/NHibernate/AdoNet/ConnectionManager.cs

```diff
@@ -215,8 +214,9 @@ public DbConnection GetConnection()
 				if (_ownConnection)
 				{
 					_connection = Factory.ConnectionProvider.GetConnection();
+					if (_currentSystemTransaction != null)
+						_connection.EnlistTransaction(_currentSystemTransaction);
```

Comment:
Now trust transaction factory for having set/unset the transaction as required, instead of seeking itself the current one (and eventually enlisting with a sub-scope instead of the one in which is the session).